test/model.js

```diff
@@ -580,6 +618,25 @@
     equal(newModel.destroy(), false);
   });
 
+  test("destroy will pass extra options to success callback", 1, function () {
+    var SpecialSyncModel = Backbone.Model.extend({
+      sync: function (method, model, options) {
+        _.extend(options, { specialSync: true });
+        return Backbone.Model.prototype.sync.call(this, method, model, options);
+      },
+      urlRoot: '/test'
+    });
+
+    var model = new SpecialSyncModel({ id: 'id' });
+
+    model.destroy({ success: onSuccess });
+    this.ajaxSettings.success();
+
+    function onSuccess(model, response, options) {
```

Comment:
Not even in a test?

This is an asynchronous bit, but in order for the test to read more naturally, I always like to put assertions after actions. This technique takes advantage of function hoisting to allow this to work. `var onSuccess = function() { /* ... */ }` will not work-- the declaration is hoisted, but the assignment doesn't happen until the line is reached, so I would have to move that function up. I would never do this in actual code.

That said, if there's no flexibility forthcoming on this, I can certainly make some tweaks. I just wanted to explain my rationale first and make sure it was understood I didn't do it randomly and/or without knowing the potential dangers.